### PR TITLE
Add an option to set the deployment action cache TTL

### PR DESCRIPTION
**Pull request by lburgazzoli**
## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pkg/controller/actions/deploy/action_deploy_cache.go [hunk 1]
```diff
@@ -10,27 +10,44 @@ import (
...
+	}
+}
+
+func newCache(opts ...CacheOpt) *Cache {
+	c := Cache{
+		ttl: DefaultCacheTTL,
+	}
+
+	for i := range opts {
+		opts[i](&c)
```
Comment (ykaliuta) [inline pkg/controller/actions/deploy/action_deploy_cache.go:35]: Wondering, why you are iterating over indexes vs values.  To avoid aliasing just in case?

**Comment by ykaliuta** [comment]
lgtm

File: pkg/controller/actions/deploy/action_deploy_cache.go (hunk 1 above)
Comment (lburgazzoli) [reply pkg/controller/actions/deploy/action_deploy_cache.go]: done

Comment (lburgazzoli) [reply pkg/controller/actions/deploy/action_deploy_cache.go]: changed it anyway :)

**Review by zdtsw — APPROVED** [review]

**Pull request merged**